internal/provider/template.go

```diff
@@ -245,13 +245,15 @@ Import is supported using the following syntax:
 
 const defaultProviderTemplate providerTemplate = `---
 ` + frontmatterComment + `
-page_title: "{{.ProviderShortName}} Provider"
+{{- $name := .ProviderShortName}}
+{{- if ne .ProviderName .RenderedProviderName }}{{ $name = .RenderedProviderName}}{{ end }}
```

Comment:
i believe the logic is incorrect here as this will always attempt to use `.RenderedProviderName` since `.ProviderName` is always going to differ from `.RenderedProviderName`.

i think what want is more like this:

```
{- $name := .ProviderShortName}}
{{- if .RenderedProviderName }}{{ $name = .RenderedProviderName }}{{ end }}
page_title: "{{ $name }} Provider"
```

Reply:
I think `.RenderedProviderName` is always set, and its default value is `.ProviderName`. Here's the reference:
https://github.com/hashicorp/terraform-plugin-docs/blob/2f535d35126c84704b8fe27e197a61db51d6f8c7/internal/provider/generate.go#L123-L125
Feel free to try it out!

Reply:
@jacobbednarz have you had the chance to check again?

Reply:
i haven't, however, being explicit in the template makes this clear what the expected outcome is instead of relying on overrides within the generation process.

Reply:
I totally agree! Actually, the solution you proposed was the first one that came to my mind as well. Unfortunately, it just doesn't work because the `.RenderedProviderName` is never unset, as shown in the previous comment.
Thus, this PR will keep the default behaviour the same as it was before, and it overrides the provider name in the template only if the specific variable has been set.